### ISSUE TITLE
Providing a class method as a target for builds

### ISSUE THREAD

**Issue by jgbos** [mit-ll-responsible-ai/hydra-zen#133]
Error:

```
ImportError: Error instantiating 'hydra_zen.funcs.zen_processing' : Encountered error: `No module named 'pytorch_lightning.core.datamodule.from_datasets'; 'pytorch_lightning.core.datamodule' is not a package` when loading module 'pytorch_lightning.core.datamodule.from_datasets'
```

To reproduce:

```python
from hydra_zen import builds, MISSING
from pytorch_lightning import LightningDataModule
from torchvision import datasets

CIFAR10 = builds(datasets.CIFAR10,  root=MISSING,  download=True)

CIFAR10DataModule = builds(
    LightningDataModule.from_datasets,
    num_workers=4,
    batch_size=256,
    train_dataset=CIFAR10(root="${..data_dir}"),
    zen_meta=dict(data_dir=MISSING)
)

instantiate(CIFAR10DataModule, data_dir="/path/to/cifar10")
```

**edit**

This does not appear to be a `zen_processing` issue, but either a Hydra or hydra-zen issue.  Removing `zen_meta` above provides the following error:

```
Traceback (most recent call last):
  File "/home/justin_goodwin/projects/raiden/hydra-zen-example/image_classifier/image_classifier/main.py", line 38, in main
    data = instantiate(cfg.experiment.data)
  File "/home/justin_goodwin/.conda/envs/raiden/lib/python3.8/site-packages/hydra/_internal/instantiate/_instantiate2.py", line 180, in instantiate
    return instantiate_node(config, *args, recursive=_recursive_, convert=_convert_)
  File "/home/justin_goodwin/.conda/envs/raiden/lib/python3.8/site-packages/hydra/_internal/instantiate/_instantiate2.py", line 240, in instantiate_node
    _target_ = _resolve_target(node.get(_Keys.TARGET))
  File "/home/justin_goodwin/.conda/envs/raiden/lib/python3.8/site-packages/hydra/_internal/instantiate/_instantiate2.py", line 104, in _resolve_target
    return _locate(target)
  File "/home/justin_goodwin/.conda/envs/raiden/lib/python3.8/site-packages/hydra/_internal/utils.py", line 587, in _locate
    raise ImportError(
ImportError: Encountered error: `No module named 'pytorch_lightning.core.datamodule.from_datasets'; 'pytorch_lightning.core.datamodule' is not a package` when loading module 'pytorch_lightning.core.datamodule.from_datasets'
```

**edit edit**

An issue with `classmethods`:

```
>>> print(to_yaml(builds(LightningDataModule)))
_target_: pytorch_lightning.core.datamodule.LightningDataModule

>>> print(to_yaml(builds(LightningDataModule.from_datasets)))
_target_: pytorch_lightning.core.datamodule.from_datasets
```

**Comment by rsokl** [comment]
Yep, this is an issue with providing a classmethod to builds. It occurs to me that we can actually accommodate via `zen_processing`. We can use the process laid out [here](https://stackoverflow.com/a/19228282) to see if the provided target is actually a class method. And if it is, we can specify the following yam;:

```
_target_ = hydra_zen.funcs.zen_processing
_zen_target = LightningDataModule
_zen_method = from_datasets
```

And then `zen_processing` can be updated to handle `_zen_method` via something like:

```python
target_parent = get_obj(_zen_target)  # import class
target = getattr(target_parent, _zen_method)  # get class method

return target(*args, **kwargs)  # instantiate class method
```

**Comment by rsokl** [comment]
Update:

Actually, we totally did support class methods in `v0.2.0`!!

```python
class C:
    @classmethod
    def f(cls):
        return 1

>>> instantiate(builds(C.f))
1
```

So https://github.com/mit-ll-responsible-ai/hydra-zen/pull/116 actually was a compatibility-breaking PR. This is pretty nasty for anyone who had been builds-ing a class method! 

Changing this to a `v0.3.0` milestone and categorizing it as a bugfix.

I simply did not realize that Hydra supported this to begin with. The good new is that, because Hydra *does* support this, this will be a much simpler fix on our end than I had described above.
